rotkehlchen/api/v1/fields.py

```diff
@@ -407,10 +400,7 @@ def _deserialize(
                 if real_value.startswith(NFT_DIRECTIVE):
                     asset = Nft(identifier=real_value)
                 else:
-                    # Just to check identifier's existence
-                    asset = Asset(identifier=real_value).resolve_to_asset_with_name_and_type()
-            elif self.expected_type == AssetWithNameAndType:
-                asset = Asset(identifier=real_value).resolve_to_asset_with_name_and_type()
+                    asset = Asset(identifier=real_value).check_existence()
```

Comment:
This thing here is needed since we check in one resource that the type is `AssetWithNameAndType`. If you change this then the resource schema needs to change.

A test will fail

Reply:
Hm.... then I am curious why linting has not failed previously? I removed this case since there is no such possible value like `AssetWithNameAndType` for `expected_type`.

Is it due to `AssetWithNameAndType` is a child of `Asset`, and thus passes? I'll check and improve the typing there then